test-data/unit/parse-errors.test

```diff
@@ -287,210 +260,189 @@ def g(*x): # type: (X) -> Y
 
 [case testYieldFromAfterReturn]
 def f():
     return yield from h()
 [out]
-file:2: error: Parse error before "yield"
+file:2: error: invalid syntax
 
 [case testImportDotModule]
 import .x
 [out]
-file:1: error: Parse error before .
+file:1: error: invalid syntax
 
 [case testImportDot]
 import .
 [out]
-file:1: error: Parse error before .
+file:1: error: invalid syntax
 
 [case testInvalidFunctionName]
 def while(): pass
 [out]
-file:1: error: Parse error before "while"
-file:1: error: Parse error before end of line
+file:1: error: invalid syntax
 
 [case testInvalidEllipsis1]
 ...0
 ..._
 ...a
 [out]
-file:1: error: Parse error before numeric literal
-file:2: error: Parse error before "_"
-file:3: error: Parse error before "a"
+file:1: error: invalid syntax
 
 [case testBlockStatementInSingleLineIf]
 if 1: if 2: pass
 [out]
-file:1: error: Parse error before "if"
+file:1: error: invalid syntax
 
 [case testBlockStatementInSingleLineIf2]
 if 1: while 2: pass
 [out]
-file:1: error: Parse error before "while"
+file:1: error: invalid syntax
 
 [case testBlockStatementInSingleLineIf3]
 if 1: for x in y: pass
 [out]
-file:1: error: Parse error before "for"
+file:1: error: invalid syntax
 
 [case testUnexpectedEllipsis]
 a = a...
 [out]
-file:1: error: Parse error before ...
-
-[case testFromFutureImportStar]
-from __future__ import *
-x = 1
-[out]
-file:1: error: Parse error before *
+file:1: error: invalid syntax
 
 [case testParseErrorBeforeUnicodeLiteral]
 x u'y'
 [out]
-file:1: error: Parse error before string literal
+file:1: error: invalid syntax
 
 [case testParseErrorInExtendedSlicing]
 x[:,
 [out]
-file:1: error: Parse error before end of line
+file:1: error: unexpected EOF while parsing
 
 [case testParseErrorInExtendedSlicing2]
 x[:,::
 [out]
-file:1: error: Parse error before end of line
+file:1: error: unexpected EOF while parsing
 
 [case testParseErrorInExtendedSlicing3]
 x[:,:
 [out]
-file:1: error: Parse error before end of line
+file:1: error: unexpected EOF while parsing
 
 [case testPython2OctalIntLiteralInPython3]
 0377
 [out]
-file:1: error: Invalid numeric literal
+file:1: error: invalid token
 
 [case testInvalidEncoding]
 # foo
 # coding: uft-8
 [out]
-file:2: error: Unknown encoding 'uft-8'
+file:0: error: unknown encoding: uft-8
 
 [case testInvalidEncoding2]
 # coding=Uft.8
 [out]
-file:1: error: Unknown encoding 'Uft.8'
+file:0: error: unknown encoding: Uft.8
 
 [case testInvalidEncoding3]
 #!/usr/bin python
 # vim: set fileencoding=uft8 :
 [out]
-file:2: error: Unknown encoding 'uft8'
+file:0: error: unknown encoding: uft8
 
 [case testDoubleEncoding]
 # coding: uft8
 # coding: utf8
 # The first coding cookie should be used and fail.
 [out]
-file:1: error: Unknown encoding 'uft8'
+file:0: error: unknown encoding: uft8
 
 [case testDoubleEncoding2]
 # Again the first cookie should be used and fail.
 # coding: uft8
 # coding: utf8
 [out]
-file:2: error: Unknown encoding 'uft8'
-
-[case testDoubleEncoding3]
```

Comment:
I think the "portability issues" reason is a feeble excuse, UTF-8 as the default for Python 3 is universal. More likely either there's some code somewhere that doesn't encode/decode right, or it's hard to edit files with UTF-8 in them (I can agree with the latter). But it's fine.